monai/transforms/spatial/array.py

```diff
@@ -425,16 +432,28 @@ def __call__(
             data_array (resampled into `self.pixdim`), original affine, current affine.
 
         """
-        sr = int(data_array.ndim - 1)
+        # if the input isn't MetaTensor and output isn't desired to be one either,
+        # nothing to do.
+        if not isinstance(data_array, MetaTensor) and not get_track_meta():
+            warnings.warn(
+                "`Orientation` applied to non-MetaTensor object and metadata tracking is off, transform does nothing."
+            )
```

Comment:
I think this checking should be a utility function so it'd be easier to maintain